extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1034,24 +1034,9 @@ protected void tryInit()
 
   private void updatePartitionDataFromKafka()
   {
-    Map<String, List<PartitionInfo>> topics;
-    try {
-      synchronized (consumerLock) {
-        topics = consumer.listTopics(); // updates the consumer's list of partitions from the brokers
-      }
-    }
-    catch (Exception e) { // calls to the consumer throw NPEs when the broker doesn't respond
-      log.warn(
-          e,
-          "Unable to get partition data from Kafka for brokers [%s], are the brokers up?",
-          ioConfig.getConsumerProperties().get(KafkaSupervisorIOConfig.BOOTSTRAP_SERVERS_KEY)
-      );
-      return;
-    }
-
-    List<PartitionInfo> partitions = topics.get(ioConfig.getTopic());
+    List<PartitionInfo> partitions = consumer.partitionsFor(ioConfig.getTopic());
```

Comment:
Why did the lock and exception handling get removed? The lock seems needed, and checking the docs `partitionsFor` can throw a handful of exceptions too:
```
Throws:
WakeupException - if wakeup() is called before or while this function is called
InterruptException - if the calling thread is interrupted before or while this function is called
AuthorizationException - if not authorized to the specified topic
TimeoutException - if the topic metadata could not be fetched before expiration of the configured request timeout
KafkaException - for any other unrecoverable errors
```